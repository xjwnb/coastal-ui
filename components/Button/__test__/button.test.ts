import { mount } from "@vue/test-utils";
import { test, expect, it } from "vitest";
import Button from "../src/button.vue";

test("mount component", async () => {
  expect(Button).toBeTruthy();

  const wrapper = mount(Button, {
    props: {
      type: "info",
    },
  });

  // expect(wrapper.text()).toContain('4 x 2 = 8')
  expect(wrapper.attributes()).toEqual({ class: "ct-button ct-button-info" });

  await wrapper.trigger('click')
});
