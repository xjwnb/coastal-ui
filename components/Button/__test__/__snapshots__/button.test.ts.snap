// Vitest Snapshot v1

exports[`mount component 1`] = `"<div class=\\"ct-button\\"><span></span></div>"`;
